### PR TITLE
Add VS2022 solution

### PR DESCRIPTION
**Pull request by NorthFury**
I removed the already committed user local project settings before adding a rule for them to gitignore.
The `ext\VFS\build\` path was ignored and shouldn't be. I updated the gitignore file to be more specific in regards to the build directories.
I added a VS2022 solution that builds to VS2022 instead of VS2013.

### CODE REVIEW

**Comment by Asdow** [comment]
We've switched over to cmake, so the solution files are no longer necessary IIRC.
@majcosta was this correct?

**Comment by majcosta** [comment]
> We've switched over to cmake, so the solution files are no longer necessary IIRC. @majcosta was this correct?

Yes. I am very overdue with the PR to clean them up, sorry.

File: .gitignore [hunk 1]
```diff
@@ -4,8 +4,10 @@ cmake-build-*/
 
 # Visual Studio 2022
 .vs/
-build/
-lib/
+/build/
+/bin/
+/lib/
```
Comment (majcosta) [inline .gitignore:9]: these two aren't needed anymore, if you want to remove them while you're at it I'd be grateful

File: .gitignore [hunk 2]
```diff
@@ -4,8 +4,10 @@ cmake-build-*/
...
 out/
 CMakeSettings.json
 /CMakeUserPresets.json
+*.vcxproj.user
```
Comment (majcosta) [inline .gitignore:13]: same as above

**Review by majcosta — CHANGES_REQUESTED** [review]
first two commits are great, thank you! last one can be dropped, sorry for your trouble

**Comment by NorthFury** [comment]
I'm closing this PR since we're removing all visual studio solution and project files. I can open up a new one if there will be any changes still needed.

**Comment by majcosta** [comment]
Like I said, the first two commits are a good change. There's no need for a new PR, you can just drop the `add vs2022 solution` and update this one.

If you're unsure how to do it, this is a good StackOverflow answer: https://stackoverflow.com/a/51400593

**Comment by NorthFury** [comment]
I know how to use git. The main purpose of this PR was to add a VS22 solution. Every other change was supporting that purpose and ensuring there is a good dev experience with it.

For the changes you mentioned to keep:
* you are already deleting the user project files in your PR.
* most of my changes for the gitignore file are obsolete already

I'll look over how the new dev experience is and update the gitignore file afterwards to something that fits the needs.

**Pull request closed without merging**